[DOCS-7210] Datadog administrator's guide (#24988)
DataDog/documentation: content/en/administrators_guide/build.md

---
title: Build your Datadog installation

> description: Setup and install Datadog as an administrator to avoid pit-falls down the road

Reviewer: ```suggestion
description: Setup and install Datadog as an administrator to avoid pitfalls down the road
```

Reviewer: I feel that each page should have a different description, but that can be done in a future update.

further_reading:
- link: "/getting_started/integrations/"
  tag: "Documentation"
  text: "Getting Started with Integrations"
---

After you plan your Datadog installation design and best practices, concentrate on the construction of Datadog itself, understanding what needs to be installed, and the best way to achieve that.

As you scale your IT footprint, it's important to develop standards, methods, and practices for how software is installed and used. Engineering effort is dedicated to developing precise steps for installing and configuring software in a repeatable, standardized manner, while still allowing for a significant level of flexibility. Without these standards, reliability can be impacted. In this section, we describe how Datadog can most efficiently interact with those standards, methods, and practices.

## Iterating on your environment

> In the [plan][7] section, you explored a range of topics within a Datadog design specification. In a perfect world, every one of those questions would be fully researched and answered before executing a large roll-out. In this regard, enterprise IT engineering is far from a perfect world, and sometimes you need to pause and adapt as you build out your installation.

Reviewer: ```suggestion
In the [plan][7] section, you explored a range of topics within a Datadog design specification. In a perfect world, every one of those questions would be fully researched and answered before executing a large rollout. In this regard, enterprise IT engineering is far from a perfect world, and sometimes you need to pause and adapt as you build out your installation.
```

All the objects you create in Datadog, such as the dashboards, alerts, notebooks, parsed logs, and configurations for cloud integrations are stored in the platform as JSON. They are exportable and importable. This opens a host of administration capabilities, including Full Infrastructure as Code (IaC) compliance, configuration backup, account migration, and reusability. Datadog also supports a [Terraform Provider][2], and a [CLI Tool][3] for these purposes.

## Provisioning

At the core of every enterprise IT environment lies a provisioning process, the "means of production" for the various software titles and instances that make up an IT architecture.
To successfully manage Datadog at scale, integrate it into your enterprise's provisioning process. There are many ways to accomplish this, thanks to the simple installation model of the Datadog Agent.   

### Modular architecture

Like most enterprise software products, Datadog installations can be broken into three separate operations, each a part of the [modular architecture][6] referred to as the file/package/service model.

**File(s)**: Contains configurations  
**Package**: Holds binaries and controls their deployment  

> **Service**: Manages the runtime instance via the OS service system.  

Reviewer: ```suggestion
**Service**: Manages the runtime instance via the OS service system
```

The basic operations you must complete in order to install Datadog are the following:

**File**: Source code control can be used to store and control the edits of configuration files. Templating and IaC solutions like Jinja2 and Ansible are also highly effective.

**Package**: Use internal package repositories such as Artifactory or Nexus to host .rpm, .msi, and containerized Agent packages.   

> **Service**: Use of IaC, or shell scripting

Reviewer: ```suggestion
**Service**: Use of IaC, or shell scripting.
```

**IaC:** Infrastructure-As-Code has advanced in both sophistication and robustness. While it is almost universally used in cloud infrastructures, it is often retrofitted to long-established on-premise infrastructures. Its simple file/package/service structure has been leveraged to deploy significant Datadog footprints with IaC "tools" as rudimentary as a bash script. While this is not recommended, it stands as encouragement to begin the IaC adoption of Datadog as soon as possible, and when you do, you will find Datadog at the ready with sample code and integrations for Ansible, Puppet, Chef, Powershell, Bash, CloudFormations, Terraform, and more.  

**Recommendations:**   

> When it comes to deploying Datadog Agent software, it is advisable to re-use as much of your existing provisioning systems as possible. Datadog software design is flat and compliant to industry standard methods.  

Reviewer: ```suggestion
When it comes to deploying Datadog Agent software, it is advisable to reuse as much of your existing provisioning systems as possible. Datadog software design is flat and compliant to industry standard methods.  
```
